### ISSUE TITLE
Make all video type checks case-insensitive

### ISSUE THREAD

**Issue by foliovision** [foliovision/freedom-video-player#480]
Original issue link: [#661 Make all video type checks case-insensitive](https://github.com/flowplayer/flowplayer/issues/661)

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Apr 17, 2014

Typical example: `application/x-mpegURL`.
Every time we parse an actual video tag we should set `type.toLowerCase()` for internal handling and comparison.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Apr 17, 2014

And we should make the source type parsing tests case-insensitive too.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/265438?v=4' /> [nnarhinen](https://github.com/nnarhinen) commented on Jul 29, 2014

This is handled by the now merged pull request, right?

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Jul 29, 2014

https://github.com/flowplayer/flowplayer/blob/dev/lib/engine/flash.js#L22 would be another corner-case candidate.
And maybe even https://github.com/flowplayer/flowplayer/blob/dev/lib/engine/html5.js#L43

According to http://www.w3.org/Protocols/rfc1341/4_Content-Type.html
"The type, subtype, and parameter names are not case sensitive."

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Jul 30, 2014

It certainly won't do any harm. Good.
I grepped for more potential candidates and did not find any (probably only after release ;-))
